app/controllers/trainees/personal_details_controller.rb

```diff
@@ -1,8 +1,40 @@
 module Trainees
   class PersonalDetailsController < ApplicationController
+    before_action :build_nationalities, only: %w[edit update]
+
     def edit
-      @trainee = Trainee.find(params[:id])
+      @personal_details_form = PersonalDetailsForm.new(trainee: trainee)
+    end
+
+    def update
+      @personal_details_form = PersonalDetailsForm.new(trainee: trainee, params: personal_details_params)
+
+      if @personal_details_form.submit
+        redirect_to trainee_path(@personal_details_form.trainee)
+      else
+        render :edit
+      end
+    end
+
+  private
+
+    def trainee
+      @trainee ||= Trainee.find(params[:trainee_id])
```

Comment:
Should we raise an error if not trainee_id was passed in?

Reply:
We did that somewhere else but now I think about it it'll 404 before it gets here as it's part of the URL. Unless it was an empty segment in the URL I suppose but it would still 404 when this was called with nil I think.